src/Fixer/Whitespace/BlankLineBeforeControlStatementFixer.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Andreas Möller <am@localheinz.com>
+ */
+final class BlankLineBeforeControlStatementFixer extends AbstractFixer implements ConfigurableFixerInterface, WhitespacesAwareFixerInterface
+{
+    /**
+     * @var array
+     */
+    private static $defaultConfiguration = array(
+        'return',
+    );
+
+    /**
+     * @var array
+     */
+    private static $tokenMap = array(
+        'break' => T_BREAK,
+        'continue' => T_CONTINUE,
+        'declare' => T_DECLARE,
+        'do' => T_DO,
+        'else' => T_ELSE,
+        'elseif' => T_ELSEIF,
+        'for' => T_FOR,
+        'foreach' => T_FOREACH,
+        'if' => T_IF,
+        'include' => T_INCLUDE,
+        'include_once' => T_INCLUDE_ONCE,
+        'require' => T_REQUIRE,
+        'require_once' => T_REQUIRE_ONCE,
+        'return' => T_RETURN,
+        'switch' => T_SWITCH,
+        'throw' => T_THROW,
+        'try' => T_TRY,
```

Comment:
👍 for `catch` and `finally`, they are quite similar to `else` or `elseif` so why not support them as well?